### ISSUE TITLE
Broadcast: Cannot read properties of undefined (reading 'Sender') on ws_1.WebSocket.Sender

### ISSUE THREAD

**Issue by offaxis** [socketio/socket.io-redis-adapter#478]
Hi !

I used to broadcast socket and worked fine ! But a new error appeared on calling `client.broadcast.emit('hello');`

Error:
```
/.../node_modules/socket.io-adapter/dist/index.js:170
            packetOpts.wsPreEncodedFrame = ws_1.WebSocket.Sender.frame(data, {
                                                          ^

TypeError: Cannot read properties of undefined (reading 'Sender')
    at RedisAdapter._encode (/.../node_modules/socket.io-adapter/dist/index.js:170:59)
    at RedisAdapter.broadcast (/.../node_modules/socket.io-adapter/dist/index.js:117:37)
    at RedisAdapter.broadcast (/.../node_modules/@socket.io/redis-adapter/dist/index.js:466:15)
    at BroadcastOperator.emit (/.../node_modules/socket.io/dist/broadcast-operator.js:169:26)
    at Socket.<anonymous> (/.../server/api/user/user.socket.js:19:34)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    at Socket.emitUntyped (/.../node_modules/socket.io/dist/typed-events.js:69:22)
    at /.../node_modules/socket.io/dist/socket.js:614:39
    at processTicksAndRejections (node:internal/process/task_queues:77:11)
```

If i remove the broadcast instruction like `client.emit('hello');` it works! But the current client received it too...

Versions:
    "@socket.io/redis-adapter": "^8.0.0", // same issue with v7
    "socket.io": "^4.5.4",
    "socket.io-client": "^4.5.4",

**Comment by darrachequesne** [comment]
I'm digging into this.

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-adapter/commit/99b0f188194b58a213682d564607913a447279e3, included in `socket.io-adapter@2.5.1`. Could you please check by running `npm update --depth 2 socket.io-adapter`?

**Comment by offaxis** [comment]
Thanks for the quick answer !!

It seems there is an error:
```
Error: Cannot find module 'ws'
Require stack:
- /usr/src/app/node_modules/socket.io-adapter/dist/index.js
- /usr/src/app/node_modules/@socket.io/redis-adapter/dist/index.js
- /usr/src/app/dist/server.bundle.js
````

**Comment by darrachequesne** [comment]
Hmm, that's weird, `ws` is listed as a [peer dependency](https://github.com/socketio/socket.io-adapter/blob/4e3991f740808afc91eb7b12b8845595cc3a158d/package.json#L16), and is transitively imported by the `socket.io` package.

- which version of Node.js are you using?
- which OS?
- did you try to remove the `node_modules` directory and reinstall all modules?

**Comment by offaxis** [comment]
which version of Node.js are you using?
=> 18 (Docker image node:18-alpine)

which OS?
=> on local : no error but on prod server the problem appeared (AWS ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-20200430)

did you try to remove the node_modules directory and reinstall all modules?
=> yes but same

I have listed on my project dependencies `socket.io-adapter ^2.5.1` maybe it is the problem ?

**Comment by offaxis** [comment]
Hi !

We found the bug: we use the command `npm install --omit=dev` to install project, which do not include the `ws` dependency witch is listed in `peerDependecies` in [`socket.io-adapter`](https://github.com/socketio/socket.io-adapter/blob/main/package.json)

So to fix this, i added the `ws` lib manually in my production dependencies.

Thanks for your work by the way!

**Comment by darrachequesne** [comment]
Unfortunately, I'm not able to reproduce the issue:

```
$ node -v
v18.12.1

$ npm install --omit=dev socket.io @socket.io/redis-adapter redis

added 26 packages, and audited 27 packages in 742ms

found 0 vulnerabilities

$ npm ls ws
tmp@ /tmp
├─┬ @socket.io/redis-adapter@8.0.0
│ └─┬ socket.io-adapter@2.5.1
│   └── ws@8.12.0
└─┬ socket.io@4.5.4
  └─┬ engine.io@6.2.1
    └── ws@8.2.3
```

I guess we can list `ws` in `dependencies` instead of `peerDependencies`, in order to prevent this issue, but I don't understand why it's not working as expected.

Related: https://github.com/socketio/socket.io-redis-adapter/issues/481

**Comment by offaxis** [comment]
We use `npm v8`, it seems to be important.

The manual installation of `ws` is a patch, to unblock the situation, but we hope, it will be added as `dependencies` in `socket.io-adapter`

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-redis-adapter/commit/7aecf01c361602f17ee3759da1ceeabb460d4519, included in version [8.0.1](https://github.com/socketio/socket.io-redis-adapter/releases/tag/8.0.1).
